### DISCUSSION TITLE
[SOS] milvus queryNode不停的在重启

### DISCUSSION THREAD

**Discussion by RongHouchao** [milvus-io/milvus#21567] [Q&A and General discussion]
milvus集群中存了80万数据后，queryNode一直在重启。
![image](https://user-images.githubusercontent.com/117245915/210955243-c31cf382-edf6-4500-8be0-700e0c6a13b6.png)

日志中也没有报错信息，服务就莫名其妙的死掉了。
每次日志都是同样的结束位置。
![image](https://user-images.githubusercontent.com/117245915/210955557-1c6aaf2c-d6e6-485e-9053-e24f07d6e95f.png)

一直在报GC，是queryNode分配的资源不够吗？
使用的是milvus-helm的master分支部署的，我看这上面的value.yaml中没有配置queryNode资源限制的配置项

**Comment by xiaofan-luan** [comment]
有检查过K8s整体的内存么？看情况像是OOM kill

**Reply by RongHouchao** [reply]
看过内存了，整体内存还有 2G+ 的空余，但是这个pod就是一直在重启。这个queryNode有设置资源占用的地方吗？

**Reply by xiaofan-luan** [reply]
是单机版么？物理机内存还有2G

**Reply by xiaofan-luan** [reply]
现在又确定挂掉的原因么？可以尝试开一下coredump

**Reply by RongHouchao** [reply]
物理机内存还有2G，不是单机版的，是集群版的。使用的是milvus-helm的master分支部署的。
没找到挂掉的原因，看日志是一直在gc。
您说的这个coredump是queryNode的coredump吗，这个怎么开

**Reply by xiaofan-luan** [reply]
> 物理机内存还有2G，不是单机版的，是集群版的。使用的是milvus-helm的master分支部署的。 没找到挂掉的原因，看日志是一直在gc。 您说的这个coredump是queryNode的coredump吗，这个怎么开

那个日志应该只是一次GC，不影响具体逻辑，所以log这里什么都没有？
Dmesg有信息么

**Reply by RongHouchao** [reply]
我试了几次，找了一些规律。
存数据到70万的时候queryNode会挂，但是如果milvus集群所在的node重启一下，milvus集群就正常了。
我尝试了70万数据、80万数据、100万数据，都是这种情况，node一重启就可以提供服务了。

**Comment by LoveEachDay** [comment]
@RongHouchao Could run `kubectl describe pods <query-node>` to show the restart reason?

**Reply by RongHouchao** [reply]
我试了几次，找了一些规律。
存数据到70万的时候queryNode会挂，但是如果milvus集群所在的node重启一下，milvus集群就正常了。
我尝试了70万数据、80万数据、100万数据，都是这种情况，node一重启就可以正常提供服务了。
